src/main/kotlin/no/nav/familie/ef/sak/iverksett/oppgaveforbarn/BarnFyllerÅrOppfølgingsoppgaveService.kt

```diff
@@ -0,0 +1,142 @@
+    private val personopplysningerIntegrasjonerClient: PersonopplysningerIntegrasjonerClient
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+    private val secureLogger = LoggerFactory.getLogger("secureLogger")
+
+    fun opprettOppgaverForAlleBarnSomHarFyltÅr(dryRun: Boolean = false) {
+        val dagensDato = LocalDate.now()
+        val alleBarnIGjeldendeBehandlinger =
+            gjeldendeBarnRepository.finnBarnAvGjeldendeIverksatteBehandlinger(StønadType.OVERGANGSSTØNAD, dagensDato) +
+                gjeldendeBarnRepository.finnBarnTilMigrerteBehandlinger(StønadType.OVERGANGSSTØNAD, dagensDato)
+
+        logger.info("Antall barn i gjeldende behandlinger: ${alleBarnIGjeldendeBehandlinger.size}")
+
+        val skalOpprettes = filtrerBarnSomHarFyltÅr(alleBarnIGjeldendeBehandlinger)
+
+        if (!dryRun) {
+            opprettOppgaveForBarn(skalOpprettes)
+        } else {
+            logger.info("Ville opprettet oppgave for ${skalOpprettes.size} barn.")
+            skalOpprettes.forEach {
+                secureLogger.info(
+                    "Ville opprettet oppgave for barn med fødselsnummer: " +
+                        "${it.fødselsnummer} med alder ${it.alder}"
+                )
+            }
+        }
+    }
+
+    private fun filtrerBarnSomHarFyltÅr(barnTilUtplukkForOppgave: List<BarnTilUtplukkForOppgave>): List<OpprettOppgaveForBarn> {
+        val opprettedeOppgaver = oppgaveRepository.findByTypeAndAlderIsNotNull(Oppgavetype.InnhentDokumentasjon)
+        val skalOpprettes = mutableListOf<OpprettOppgaveForBarn>()
+
+        barnTilUtplukkForOppgave.forEach { barn ->
+            val barnetsAlder = Alder.fromFødselsdato(fødselsdato(barn))
+            if (barnetsAlder != null && barn.fødselsnummerBarn != null && opprettedeOppgaver.none { it.barnPersonIdent == barn.fødselsnummerBarn && it.alder == barnetsAlder }) {
+                skalOpprettes.add(
+                    OpprettOppgaveForBarn(
+                        barn.fødselsnummerBarn,
+                        barn.fødselsnummerSøker,
+                        barnetsAlder,
+                        barn.behandlingId
+                    )
+                )
+            }
+        }
+
+        logger.info("barn til utplukk for oppgave: ${skalOpprettes.size}")
+        return skalOpprettes
+    }
+
+    private fun opprettOppgaveForBarn(opprettOppgaverForBarn: List<OpprettOppgaveForBarn>) {
+        if (opprettOppgaverForBarn.isEmpty()) return
+        val gjeldendeBarnList =
+            gjeldendeBarnRepository.finnEksternFagsakIdForBehandlingId(opprettOppgaverForBarn.map { it.behandlingId }).toSet()
+        gjeldendeBarnList.forEach { gjeldendeBarn ->
+            val opprettOppgaveForEksternId =
+                opprettOppgaverForBarn.firstOrNull { it.fødselsnummer == gjeldendeBarn.barnPersonIdent }
+            val finnesOppgave = oppgaveRepository.findByBehandlingIdAndBarnPersonIdentAndAlder(
```

Comment:
Her sjekker vi oppgave-tabellen på nytt?

Reply:
Dette blir litt rørigt

1. Før denne metoden har vi laget et utplukk på barn som skal få opprettet oppgave
2. I denne henter vi alle barn til behandlingId'er for de barn som trengte oppgave fra 1, og filtrer ut de som allerede har en oppgave

Hvorfor ikke bare opprette oppgave for de barnen som kommer inn her? 
Hvis det gjøres for å finne eksternId, så anbefaler jeg å legge til en metode på BehandlingService som returnerer eks et sett av 
```kotlin
data class BehandlingId(val id: UUID, val eksternBehandlingId: Long, val fagsakId: UUID, val eksternFagsakId: Long)
```